crates/driver/src/infra/config/file/load.rs

```diff
@@ -76,7 +84,19 @@ pub async fn load(chain: eth::ChainId, path: &Path) -> infra::Config {
                 },
                 request_headers: config.request_headers,
                 rank_by_surplus_date: config.rank_by_surplus_date,
-                fee_handler: config.fee_handler,
+                protocol_fees: ProtocolFees::new(
+                    config.fee_handler,
+                    config
+                        .protocol_fee_discounted_addresses
+                        .into_iter()
+                        .map(|protocol_fee_discounted_addresses| {
+                            (
+                                protocol_fee_discounted_addresses.address,
+                                protocol_fee_discounted_addresses.kind,
+                            )
+                        })
+                        .collect::<HashMap<_, _>>(),
```

Comment:
Convert into a `HashMap` in order to have O(1) access when checking the addresses.